docbrown/core/src/tgraph.rs

```diff
@@ -274,32 +274,32 @@ impl TemporalGraph {
         }
     }
 
-    pub(crate) fn vertex(&self, v: u64) -> Option<VertexView> {
+    pub(crate) fn vertex(&self, v: u64) -> Option<VertexRef> {
         let pid = self.logical_to_physical.get(&v)?;
         Some(match self.adj_lists[*pid] {
-            Adj::Solo(lid) => VertexView {
+            Adj::Solo(lid) => VertexRef {
                 g_id: lid,
                 pid: Some(*pid),
             },
-            Adj::List { logical, .. } => VertexView {
+            Adj::List { logical, .. } => VertexRef {
                 g_id: logical,
                 pid: Some(*pid),
             },
         })
     }
 
-    pub(crate) fn vertex_window(&self, v: u64, w: &Range<i64>) -> Option<VertexView> {
+    pub(crate) fn vertex_window(&self, v: u64, w: &Range<i64>) -> Option<VertexRef> {
         let pid = self.logical_to_physical.get(&v)?;
         let w = w.clone();
         let mut vs = self.index.range(w.clone()).flat_map(|(_, vs)| vs.iter());
```

Comment:
why not kmerge and dedup here?

Reply:
I did not write this code but having had a look at it, it is indeed terribly inefficient. However, no need for kmerge and dedup, this should be iterating over the bit sets until it finds one that contains the vertex or there are no more. This is turning that into a linear search for no reason. This should just be using `any` on the range iterator...

Reply:
I made a ticket to fix this.